### PR TITLE
Add `Stage` field to `BreachedPasswordDetection` struct

### PR DESCRIPTION
**Pull request by nialdaly**
## Description
This PR aims to contribute to the closure of https://github.com/auth0/terraform-provider-auth0/issues/386.

## Checklist
- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [ ] I have added documentation for new/changed functionality in this PR.
- [ ] All active GitHub checks for tests, formatting, and security are passing.
- [ ] The correct base branch is being used, if not `main`.

### CODE REVIEW

File: management/attack_protection.go [hunk 1]
```diff
@@ -20,10 +20,22 @@ func newAttackProtectionManager(m *Management) *AttackProtectionManager {
...
-	Method                     *string   `json:"method,omitempty"`
+	Enabled                    *bool                           `json:"enabled,omitempty"`
+	Shields                    *[]string                       `json:"shields,omitempty"`
+	AdminNotificationFrequency *[]string                       `json:"admin_notification_frequency,omitempty"`
+	Method                     *string                         `json:"method,omitempty"`
+	Stage                      *BreachedPasswordDetectionStage `json:"stage,omitempty"`
+}
+
+// BreachedPasswordDetectionStage is used to specify per-stage configuration options.
+type BreachedPasswordDetectionStage struct {
```
Comment (sergiught) [inline management/attack_protection.go:31]: 👍🏻 

**Note for posterity:** We should rename the current `Stage` to `SuspiciousIpThrottlingStage` in the future.

File: management/attack_protection.go [hunk 2]
```diff
@@ -20,10 +20,22 @@ func newAttackProtectionManager(m *Management) *AttackProtectionManager {
...
+	PreUserRegistration *BreachedPasswordDetectionPreUserRegistration `json:"pre-user-registration,omitempty"`
+}
+
+// BreachedPasswordDetectionPreUserRegistration is used to specify breached password detection
+// configuration (shields) for the sign up flow.
+type BreachedPasswordDetectionPreUserRegistration struct {
+	Shields *[]string `json:"shields,omitempty"`
```
Comment (sergiught) [inline management/attack_protection.go:38]: Let's add a comment here on what shields are.

```
	// Action to take when a breached password is detected during a signup.
	// Possible values: block, admin_notification.
```

**Review by sergiught — CHANGES_REQUESTED** [review]
Hey @nialdaly 👋🏻 Thanks for the contribution! Solid work here 💪🏻 

The only things that would be great to add in this PR as well is an updated test that includes us trying to modify the Stage field and have the test recordings updated as well. 

I can help with all of this if you enable the checkbox that allows contributors to push to your PR branch.

**Comment by nialdaly** [comment]
Hi @sergiught thanks for the feedback! Happy for you to help, I've added a test but I think I'm missing something.

**Comment by sergiught** [comment]
Hey @nialdaly, awesome, thanks for those updates. The only thing that we're missing is updating the test recordings. I'm still getting access denied when I'm trying to push to your branch. 

If you don't mind replacing the contents of https://github.com/auth0/go-auth0/blob/main/management/testdata/recordings/TestAttackProtection/Get_breached_password_detection_settings.yaml

with 

```
---
version: 1
interactions:
- request:
    body: |
      {}
    form: { }
    headers:
      Content-Type:
        - application/json
      User-Agent:
        - Go-Auth0-SDK/latest
    url: https://go-auth0-dev.eu.auth0.com/api/v2/attack-protection/breached-password-detection
    method: GET
  response:
    body: '{"enabled":true,"shields":["block","user_notification","admin_notification"],"admin_notification_frequency":["weekly","immediately","monthly","daily"],"method":"standard","stage":{"pre-user-registration":{"shields":["admin_notification","block"]}}}'
    headers:
      Content-Type:
        - application/json; charset=utf-8
    status: 200 OK
    code: 200
    duration: 1ms

```

and https://github.com/auth0/go-auth0/blob/main/management/testdata/recordings/TestAttackProtection/Update_breached_password_detection_settings.yaml

with 

```
---
version: 1
interactions:
- request:
    body: |
      {}
    form: { }
    headers:
      Content-Type:
        - application/json
      User-Agent:
        - Go-Auth0-SDK/latest
    url: https://go-auth0-dev.eu.auth0.com/api/v2/attack-protection/breached-password-detection
    method: GET
  response:
    body: '{"enabled":true,"shields":["block","user_notification","admin_notification"],"admin_notification_frequency":["weekly","immediately","monthly","daily"],"method":"standard","stage":{"pre-user-registration":{"shields":["admin_notification","block"]}}}'
    headers:
      Content-Type:
        - application/json; charset=utf-8
    status: 200 OK
    code: 200
    duration: 1ms
- request:
    body: |
      {"enabled":true,"method":"standard","stage":{"pre-user-registration":{"shields":["block"]}}}
    form: { }
    headers:
      Content-Type:
        - application/json
      User-Agent:
        - Go-Auth0-SDK/latest
    url: https://go-auth0-dev.eu.auth0.com/api/v2/attack-protection/breached-password-detection
    method: PATCH
  response:
    body: '{"enabled":true,"shields":["block","user_notification","admin_notification"],"admin_notification_frequency":["monthly","daily","weekly","immediately"],"method":"standard","stage":{"pre-user-registration":{"shields":["block"]}}}'
    headers:
      Content-Type:
        - application/json; charset=utf-8
    status: 200 OK
    code: 200
    duration: 1ms
- request:
    body: |
      {}
    form: { }
    headers:
      Content-Type:
        - application/json
      User-Agent:
        - Go-Auth0-SDK/latest
    url: https://go-auth0-dev.eu.auth0.com/api/v2/attack-protection/breached-password-detection
    method: GET
  response:
    body: '{"enabled":true,"shields":["block","admin_notification","user_notification"],"admin_notification_frequency":["weekly","daily","monthly","immediately"],"method":"standard","stage":{"pre-user-registration":{"shields":["block"]}}}'
    headers:
      Content-Type:
        - application/json; charset=utf-8
    status: 200 OK
    code: 200
    duration: 1ms
- request:
    body: |
      {"enabled":true,"shields":["block","user_notification","admin_notification"],"admin_notification_frequency":["weekly","immediately","monthly","daily"],"method":"standard","stage":{"pre-user-registration":{"shields":["admin_notification","block"]}}}
    form: { }
    headers:
      Content-Type:
        - application/json
      User-Agent:
        - Go-Auth0-SDK/latest
    url: https://go-auth0-dev.eu.auth0.com/api/v2/attack-protection/breached-password-detection
    method: PATCH
  response:
    body: '{"enabled":true,"shields":["block","user_notification","admin_notification"],"admin_notification_frequency":["weekly","immediately","monthly","daily"],"method":"standard","stage":{"pre-user-registration":{"shields":["block","admin_notification"]}}}'
    headers:
      Content-Type:
        - application/json; charset=utf-8
    status: 200 OK
    code: 200
    duration: 1ms

```

That would be much appreciated! 🙇🏻

**Comment by sergiught** [comment]
Also I'll follow up after your PR with adding a CONTRIBUTING.md and explain how those recordings work with the tests. That's something missing from our side sadly 😢

**Review by sergiught — APPROVED** [review]
Much obliged @nialdaly !

**Comment by nialdaly** [comment]
Thanks! @sergiught I've ticked the `Allow edits by maintainers` box but I'm not sure why its not working.

**Pull request merged**